google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/SpannerRpc.java

```diff
@@ -216,26 +220,22 @@ public String getNextPageToken() {
       throws SpannerException;
 
   /**
-   * Creates a new backup from the source database specified in the {@link Backup} instance.
+   * Creates a new backup from the source database specified in the {@link
+   * com.google.cloud.spanner.Backup} instance.
    *
-   * @param instanceName the name of the instance where the backup should be created.
-   * @param backupId the id of the backup to create.
-   * @param backup the backup to create. The database and expireTime fields of the backup must be
-   *     filled.
+   * @param backupInfo the backup to create. The instance, database and expireTime fields of the
```

Comment:
I think we should check for these conditions in the builder for `BackupInfo` to ensure that they are actually filled. So that would mean that the `build()` method of the builder should fail if these fields have not been set.

Reply:
We can not do this in the builder itself, because there are times where the backup can be built without some of the necessary information. For instance:

- [backup.reload()](https://github.com/googleapis/java-spanner/blob/master/google-cloud-spanner/src/test/java/com/google/cloud/spanner/BackupTest.java#L199-L207)
- [backup.delete()](https://github.com/googleapis/java-spanner/blob/master/google-cloud-spanner/src/test/java/com/google/cloud/spanner/BackupTest.java#L209-L217)
- [backup.updateExpireTime()](https://github.com/googleapis/java-spanner/blob/master/google-cloud-spanner/src/test/java/com/google/cloud/spanner/BackupTest.java#L219-L229)

For this reason I moved the checks into the `DatabaseAdminClientImpl.createBackup` instead.